strawberryfields/apps/graph/max_clique.py

```diff
@@ -0,0 +1,99 @@
+largest clique in a graph. The problem is NP-Hard in a worst-case setting, which is why it is
+valuable to develop better algorithms to tackle it.
+
+The core of this module is a local search algorithm that proceeds as follows:
+
+#. A small clique in the graph is identified. The initial clique can be a single node, or it can
+   be obtained by shrinking a random subgraph, for example obtained from GBS.
+#. The clique is grown as much as possible by adding nodes connected to all nodes in the clique.
+#. When no more growth is possible, a node in the clique is swapped with a node outside of it.
+
+Steps 2-3 are repeated until a pre-determined number of steps have been completed, or until no
+more growth or swaps are possible.
+
+Summary
+-------
+
+.. autosummary::
+    local_search
+
+Code details
+------------
+"""
+import networkx as nx
+
+from strawberryfields.apps.graph import resize
+
+
+def local_search(
+    clique: list, graph: nx.Graph, iterations: int = 100, node_select: str = "uniform"
+) -> list:
+    """Use local search algorithm to identify large cliques.
+
+    This function implements a version of the local search algorithm given in
+    :ref:`pullan2006dynamic` and :ref:`pullan2006phased`. It proceeds by iteratively applying
+    phases of greedy growth and plateau search to the input clique subgraph. Growth is achieved
+    using the :func:`~strawberryfields.apps.graph.resize.clique_grow` function, which repeatedly
+    evaluates the set :math:`C_0` of nodes in the remainder of the graph that are connected to
+    all nodes in the clique, and selects one candidate node from :math:`C_0` to make a larger
+    clique. Plateau search is performed with the
+    :func:`~strawberryfields.apps.graph.resize.clique_swap` function, which evaluates the set
+    :math:`C_1` of nodes in the remainder of the graph that are connected to all but one of the
+    nodes in the clique, and selects one candidate from :math:`C_1` and its corresponding
+    unconnected node in the clique to swap.
+
+    The iterative applications of growth and search are applied until either the specified number
+    of iterations has been carried out or when a dead end has been reached. The function reaches
+    a dead end when it is unable to perform any new swaps or growth. Whenever the sets
+    :math:`C_0` and :math:`C_1` used during growth and swapping have more than one element,
+    there must be a choice of which node to add or swap. This choice is specified with the
+    ``node_select`` argument, with node selection based on uniform randomness and node degree
+    supported. Degree-based node selection involves picking the node with the greatest degree,
+    with ties settled by uniform random choice.
+
+    Args:
+        clique (list[int]): a subgraph specified by a list of nodes; the subgraph must be a clique
+        graph (nx.Graph): the input graph
+        iterations (int): number of steps in the algorithm; defaults to 100

```

Comment:
```suggestion
        iterations (int): number of steps in the algorithm
```
Since the default value is provided in the function signature, Sphinx will automatically extract it and display it in the docstring.

Reply:
Replied to this question below!

Reply:
Or above... depending on where you're reading from 😁